docs/cql/cql-extensions.md

```diff
@@ -232,6 +232,25 @@ to explicitly mark them as such.
 
 ## Expressions
 
+### NULL
+
+Scylla aims for a uniform handling of NULL values in expressions, inspired
+by SQL: The overarching principle is that a NULL signifies an absense of
+value, so any expression calculated based on a NULL also results in a NULL.
+For example, the result of `x + NULL`, `x = NULL`, `x < NULL` are all NULL,
+no matter what `x` is. Even the expression `NULL = NULL` returns NULL, not
+TRUE.
+
+From the last example we can see that the `SELECT` filter `WHERE x = NULL`
+matches (returns `TRUE`) no row. It doesn't even match rows where x is
+missing. To match rows with missing x, SQL offers the syntax `x IS NULL`
+(and similarly, also `x IS NOT NULL`). Scylla does not yet implement this
+syntax.
+
+In contrast, Cassandra is less consistent in its handling of nulls.
+The example `x = NULL` is considered an error, not a valid expression
+whose result is NULL.
+
 ## REDUCEFUNC for UDA
```

Comment:
I think it would also be good to mention that `NULLs` are handled differently in `LWT` conditions. There comparing `col = NULL` is equivalent to `COL IS NULL`

Reply:
Wait, what? Why?
How does this make sense? How can I explain this after I wrote a whole paragraph on the consistency of our approach, it turns out it's consistent - but only when it's not inconsistent? :-(

Reply:
We plan to split LWT IF to a "Cassandra compatibility dialect" and a "SQL compatible dialect" in order to resolve the conflict.